### PR TITLE
expose getBabelLoader helper

### PR DESCRIPTION
**Pull request by Guria**
Sometimes we need more then just injecting plugin.
Exposing getBabelLoader could be very helpful

### CODE REVIEW

**Comment by Guria** [comment]
Hmm. getLoader seems could be improved sligtly more. I'll post it tomorrow. Treat PR as WiP until that.

File: packages/react-app-rewired/index.js [hunk 1]
```diff
@@ -1,36 +1,25 @@
-const checkRuleLegacy = function(conf) {
-  if (!conf.loader) return false;
-  return conf.loader.indexOf("babel-loader") > -1;
-};
-
-const getLoader = function(config) {
-  let babelLoader = false;
-  let isBabel = false;
-  const checkRule = rule =>
```
Comment (Guria) [inline packages/react-app-rewired/index.js]: checkRule and checkRuleLegacy was effictively the same

File: packages/react-app-rewired/index.js [hunk 2]
```diff
@@ -1,36 +1,25 @@
...
-    rule.loader && rule.loader.indexOf("babel-loader") != -1;
-
-  config.module.rules.forEach(rule => {
-    if (!rule.oneOf) {
```
Comment (Guria) [inline packages/react-app-rewired/index.js]: rules could be nested in rule.use as well

File: packages/react-app-rewired/index.js [hunk 3]
```diff
@@ -1,36 +1,25 @@
-const checkRuleLegacy = function(conf) {
-  if (!conf.loader) return false;
-  return conf.loader.indexOf("babel-loader") > -1;
```
Comment (Guria) [inline packages/react-app-rewired/index.js]: very loose condition, eg /some-babel-loader/ will match here

File: packages/react-app-rewired/index.js [hunk 4]
```diff
@@ -1,36 +1,25 @@
...
+  var loader;
 
-      return;
-    }
-
-    isBabel = rule.oneOf.find(checkRule);
-    if (isBabel) {
-      babelLoader = isBabel;
-    }
+  rules.some(rule => {
```
Comment (Guria) [inline packages/react-app-rewired/index.js]: not sure about target node version. are we allow arrow functions?

**Comment by Guria** [comment]
Ok. Updated with improved solution.
Now it exposes generic `getLoader` helper to match loaders with any condition provided and `getBabelLoader` already with  default matcher for babel-loader.

cc greets to @amel-true from devSchacht podcast listener

**Comment by Guria** [comment]
tested on CRA 1.0.11 and below

**Comment by isubastiCadmus** [comment]
getBabelLoader will be really useful to be exposed and the injectBabelPlugin pattern is quite useful to hijack babel-loader for something like worker-loader. Looking forward for this to be merged

**Comment by timarney** [comment]
Will review soon (looks great at a glance) - have a site launching today.

**Comment by timarney** [comment]
@Guria thanks for this. 

Published react-app-rewired@1.2.1

**Pull request merged**